File: src/test/java/FileLoaderTest.java
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import util.ResourceUtils;

import java.io.IOException;
import java.util.List;

public class FileLoaderTest {

    @Test
    public void test_load_stop_words_success() throws IOException {
        ResourceUtils.populateStopWords(List.of("cat", "dog"));

        var stopWords = FileLoader.loadStopWords();

        Assertions.assertNotNull(stopWords);
        Assertions.assertEquals(2, stopWords.size());
    }

    @Test
    public void test_file_not_found() {
        Assertions.assertThrows(IOException.class, () -> {
            FileLoader.loadStopWords("nonexistent.txt");
        });

        Assertions.assertThrows(IOException.class, () -> {
            FileLoader.loadInputContent("nonexistent.txt");
        });
    }

}

File: src/test/kotlin/FileLoaderTest.kt
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class FileLoaderTest {

    @Test
    fun `File loader test`() {
        val file = FileLoader.readResourceFile("stopwords.txt")
        val stopwords = FileLoader.parseStopwordsFile(file)

        assertEquals(stopwords.size, 4)
        assertEquals(stopwords[0], "the")
        assertEquals(stopwords[1], "a")
        assertEquals(stopwords[2], "on")
        assertEquals(stopwords[3], "off")
    }
}